🐛 BUG: Wrangler not honouring `CLOUDFLARE_ACCOUNT_ID` environment variable when cached config is here

### What version of `Wrangler` are you using?

2.0.23

### What operating system are you using?

Linux

### Describe the Bug

Hello :smiley: 

When trying to publish the same directory to 2 different accounts, I get auth errors when I don't clean "cached" config: 

Simulating a publish to **abc** account  id:
```
$ cat ../node_modules/.cache/wrangler/wrangler-account.json
{ "account": { "id": "abc", "name": "" } }
```

When running wrangler with **def** account id given as environment variable:
```
$ CLOUDFLARE_ACCOUNT_ID=def CLOUDFLARE_API_TOKEN=ghi npx wrangler pages publish . --project-name=jkl --branch=mno
Retrieving cached values for account from ../node_modules/.cache/wrangler
▲ [WARNING] Warning: Your working directory is a git repo and has uncommitted changes

  To silence this warning, pass in --commit-dirty=true



✘ [ERROR] A request to the Cloudflare API (/accounts/abc/pages/projects/jkl/upload-token) failed.

  Authentication error [code: 10000]
  
  If you think this is a bug, please open an issue at:
  https://github.com/cloudflare/wrangler2/issues/new/choose
```

I suppose environment variable should override "cached" config? (if not and you consider it "by design", feel free to close this issue)

In particular, according to me, the error line `✘ [ERROR] A request to the Cloudflare API (/accounts/abc/pages/projects/jkl/upload-token) failed.` should not use account **abc** but **def** given in `CLOUDFLARE_ACCOUNT_ID`.

## Demo
I highlighted this behaviour with a CircleCI workflow:
```yml
version: 2.1
  
jobs:
  config:
    docker:
      - image: cimg/node:18.4.0
    steps:
      - run:
          name: "Publish to cloudflare pages (KO)"
          command: |
            npm i wrangler@2.0.23
            mkdir public
            cd public
            echo "Hello config" >> index.html
            mkdir -p ../node_modules/.cache/wrangler/
            echo '{ "account": { "id": "abcdefghijklmnopqrstuvwxyz", "name": "" } }' > ../node_modules/.cache/wrangler/wrangler-account.json
            CLOUDFLARE_ACCOUNT_ID=d97aaabbbcccdddaaabbbccc93a054 npx wrangler pages publish . --project-name=honour-account-id-env --branch=config
  noconfig:
    docker:
      - image: cimg/node:18.4.0
    steps:
      - run:
          name: "Publish to cloudflare pages (OK)"
          command: |
            npm i wrangler@2.0.23
            mkdir public
            cd public
            echo "Hello noconfig" >> index.html
            CLOUDFLARE_ACCOUNT_ID=d97aaabbbcccdddaaabbbccc93a054 npx wrangler pages publish . --project-name=honour-account-id-env --branch=noconfig
 
workflows:
  release_noconfig:
    jobs:
      - noconfig:
          filters:
            branches:
              only: /^main$/
  release_config:
    jobs:
      - config:
          filters:
            branches:
              only: /^main$/
```

## Failing with cached config
![ko-config](https://user-images.githubusercontent.com/104425314/182150021-140ffc73-0eec-4c36-958a-ceb630586e72.png)

## Publish without config
![ok-noconfig](https://user-images.githubusercontent.com/104425314/182150083-55ec5cd1-faf6-48e6-b5df-03388b1a976a.png)

## Related?
- #1422 
- #1585

--- a/.changeset/stupid-islands-relax.md
+++ b/.changeset/stupid-islands-relax.md
@@ -1,0 +1,5 @@
+---
+"wrangler": patch
+---
+
+Ignore cached account id when `CLOUDFLARE_ACCOUNT_ID` is specified

--- a/packages/wrangler/src/user/user.ts
+++ b/packages/wrangler/src/user/user.ts
@@ -234,6 +234,7 @@ import {
 	getCloudflareAccessToken,
 	getCloudflareAPITokenFromEnv,
 	getCloudflareGlobalAuthEmailFromEnv,
+	getCloudflareAccountIdFromEnv,
 	getCloudflareGlobalAuthKeyFromEnv,
 	getRevokeUrlFromEnv,
 	getTokenUrlFromEnv,
@@ -1096,9 +1097,10 @@ export async function getAccountId(): Promise<string | undefined> {
 
 	// check if we have a cached value
 	const cachedAccount = getAccountFromCache();
-	if (cachedAccount) {
+	if (cachedAccount && !getCloudflareAccountIdFromEnv()) {
 		return cachedAccount.id;
 	}
+
 	const accounts = await getAccountChoices();
 	if (accounts.length === 1) {
 		saveAccountToCache({ id: accounts[0].id, name: accounts[0].name });

